pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -355,19 +355,38 @@ cdef class {{name}}HashTable(HashTable):
 
         return np.asarray(locs)
 
-    def factorize(self, {{dtype}}_t values):
-        uniques = {{name}}Vector()
-        labels = self.get_labels(values, uniques, 0, 0)
-        return uniques.to_array(), labels
-
     @cython.boundscheck(False)
-    def get_labels(self, const {{dtype}}_t[:] values, {{name}}Vector uniques,
-                   Py_ssize_t count_prior, Py_ssize_t na_sentinel,
-                   object na_value=None):
+    @cython.wraparound(False)
+    def _unique_with_inverse(self, const {{dtype}}_t[:] values,
```

Comment:
minor comment: since what this algo does is what we typically call "factorize" in pandas, I would call this function "_factorize"

Reply:
@jorisvandenbossche 
Can we keep this for the follow-up please? It will make sense there, because then `factorize = unique_with_inverse(ignore_na=True)`.

Reply:
Sorry, I thought this was a reply to the other comment :-) 

> It will make sense there, because then factorize = unique_with_inverse(ignore_na=True)

Yes, and otherwise the other way around `unique(return_inverse=True) = factorize(ignore_na=False)`. The implemenation of that function is what factorize does in pandas (and not unique), so it makes sense to me to use that name.